### DISCUSSION TITLE
Feat: Handling Login Error for Unverified Users

### DISCUSSION THREAD

**Discussion by matthijs166** [payloadcms/payload#3211] [Feature Requests & Ideas]
We have implemented account verification for users, which works well. However, there is an issue when an unverified user attempts to log in without verifying their email. Currently, the system displays an “email or password provided incorrect” error, which might confuse users.

There doesn't seem to be a way to distinct the login attempt between an unverified one and an incorrect one. I have now extended the error message in the frontend that both can be possible, but it would be a nice quality of live feature within payload.

**Proposed Solution:**
Introduce a callback function that can be used to handle this specific error scenario for unverified users. Alternatively, consider returning a different error message that better indicates the unverified status of the account.

**The given error:**
`errors: [ { message: 'The email or password provided is incorrect.' } ]`

**Comment by jmikrut** [comment]
@matthijs166 — we're on this 100%

I know what you're saying and we have a proposed solution on its way shortly. I have added this to the roadmap accordingly!
